packages/front-end/components/Archetype/SimulateFeatureValues.tsx

```diff
@@ -0,0 +1,516 @@
+          environment: selectedEnvironment === "all" ? "" : selectedEnvironment,
+        }),
+      })
+        .then((data) => {
+          if (data && data.results) {
+            setFeatureResults(data.results);
+            const featureResultsMap = new Map<string, FeatureTestResult>();
+            data.results.forEach((r) => {
+              Object.keys(r).forEach((featureId) => {
+                featureResultsMap.set(
+                  featureId + r[featureId].env,
+                  r[featureId]
+                );
+              });
+            });
+            setFeatureResultsMap(featureResultsMap);
+            // keep track of what we evaluated so we don't have to do it again.
+            setEvaluatedAttributes(attributes);
+            setEvaluatedFeatures(featureItems.map((f) => f.id));
+            setEvaluatedEnvironment(selectedEnvironment);
+          }
+        })
+        .catch((e) => console.error(e));
+    }
+  }, [
+    featureItems,
+    attributes,
+    evaluatedAttributes,
+    evaluatedFeatures,
+    apiCall,
+    selectedEnvironment,
+    evaluatedEnvironment,
+  ]);
+
+  useEffect(() => {
+    refreshResults();
+  }, [featureItems, currentPage, refreshResults, selectedEnvironment]);
+
+  const archetypeMap = new Map<string, ArchetypeInterface>();
+  if (archetypes) {
+    archetypes.forEach((a) => {
+      archetypeMap.set(a.id, a);
+    });
+  }
+
+  if (loading) {
+    return <LoadingOverlay />;
+  }
+  if (!environments || environments.length === 0) {
+    return <div>No environments added</div>;
+  }
+  let attributeText = (
+    <>Select Archetype or edit user attributes to see feature results.</>
+  );
+  let attributeNodes: ReactNode[] = [];
+  if (attributes && Object.keys(attributes).length > 0) {
+    attributeText = archetype ? (
+      <>
+        Showing feature results for archetype{" "}
+        <strong>{archetypeMap.get(archetype)?.name ?? "?"}</strong>:
+      </>
+    ) : (
+      <>Showing feature results for users with attributes: </>
+    );
+    const attrsLength = Object.keys(attributes).length;
+    attributeNodes = Object.keys(attributes).map((key, i) => {
+      const attrValue = JSON.stringify(attributes[key]);
+      return (
+        <Fragment key={`attr-${key}-${i}`}>
+          <strong>{key}</strong>: <strong>{attrValue}</strong>
+          {i === attrsLength - 1 ? "" : i === attrsLength - 2 ? ", and " : ", "}
+        </Fragment>
+      );
+    });
+  }
+
+  const showEnvDropdown = true; //environments.length > maxEnvironments;
+  const numColumns = showEnvDropdown ? 4 : environments.length + 3;
+  const environmentOptions = [
+    ...environments.map((e) => {
+      return { label: e.id, value: e.id };
+    }),
+  ];
+  if (showAllEnv) {
+    environmentOptions.unshift({ label: "All", value: "all" });
+  }
+
+  const featureTableResults = (
+    <>
+      <div className="mb-3">
+        <div className="row mb-3">
+          <div className="col">
+            <div
```

Comment:
something feels a bit clunky about this component:

1. No clear CTA or button... entire div is clickable but doesn't give the impression that it is
2. Attribute values are listed in plain text on the same line, which feels hard to parse / cluttered in many cases. I think either format it in some sort of table inline or provide a tooltip or accordion to conditionally show the formatted version

Reply:
I adjusted the design to be more consistent with other sections.